src/components/application_manager/include/application_manager/hmi_capabilities.h

```diff
@@ -445,152 +438,10 @@ class HMICapabilities {
    * @param languages - the converted object
    *
    */
-  void convert_json_languages_to_obj(Json::Value& json_languages,
-                                     smart_objects::SmartObject& languages);
-
- private:
-  bool is_vr_cooperating_;
-  bool is_tts_cooperating_;
-  bool is_ui_cooperating_;
-  bool is_navi_cooperating_;
-  bool is_ivi_cooperating_;
-
-  // to check if IsReady response for corresponding interface received
-  bool is_vr_ready_response_recieved_;
-  bool is_tts_ready_response_recieved_;
-  bool is_ui_ready_response_recieved_;
-  bool is_navi_ready_response_recieved_;
-  bool is_ivi_ready_response_recieved_;
-
-  bool attenuated_supported_;
-  hmi_apis::Common_Language::eType ui_language_;
-  hmi_apis::Common_Language::eType vr_language_;
-  hmi_apis::Common_Language::eType tts_language_;
-  smart_objects::SmartObject* vehicle_type_;
-  smart_objects::SmartObject* ui_supported_languages_;
-  smart_objects::SmartObject* tts_supported_languages_;
-  smart_objects::SmartObject* vr_supported_languages_;
-  smart_objects::SmartObject* display_capabilities_;
-  smart_objects::SmartObject* hmi_zone_capabilities_;
-  smart_objects::SmartObject* soft_buttons_capabilities_;
-  smart_objects::SmartObject* button_capabilities_;
-  smart_objects::SmartObject* preset_bank_capabilities_;
-  smart_objects::SmartObject* vr_capabilities_;
-  smart_objects::SmartObject* speech_capabilities_;
-  smart_objects::SmartObject* audio_pass_thru_capabilities_;
-  smart_objects::SmartObject* pcm_stream_capabilities_;
-  smart_objects::SmartObject* prerecorded_speech_;
-  std::string ccpu_version_;
-  bool is_navigation_supported_;
-  bool is_phone_call_supported_;
-
-  ApplicationManager& app_mngr_;
-  HMILanguageHandler hmi_language_handler_;
-
-  DISALLOW_COPY_AND_ASSIGN(HMICapabilities);
+  virtual void convert_json_languages_to_obj(
+      Json::Value& json_languages, smart_objects::SmartObject& languages) = 0;
```

Comment:
This may be a const function - it does not seem to modify class fields. Also json_languages does not seem to be modified inside the method, asString() is a const method, so it may be const too.